zerver/views/user_groups.py

```diff
@@ -120,7 +120,14 @@ def edit_user_group(
     ):
         raise JsonableError(_("No new data supplied"))
 
-    user_group = access_user_group_by_id(user_group_id, user_profile, for_read=False)
+    user_group = access_user_group_by_id(
+        user_group_id, user_profile, for_read=False, allow_deactivated=True
+    )
+
+    if user_group.deactivated and (
+        description is not None or can_mention_group is not None or can_manage_group is not None
+    ):
+        raise JsonableError(_("You can only change name of deactivated user groups"))
```

Comment:
This check is fragile, in that it won't be correct when adding new settings.

One option would be to, immediately after the `name` block, place this check, so only changing `name` is permitted.

Reply:
We would need to add the `if user_group.deactivated` check in each `if description is not None`, etc. block so that we do not raise error if only name is changed. Should we do that?

Reply:
I'm not sure. Maybe best to leave this as-is for this PR, and schedule it for a follow-up. But yeah, probably the right model would be to check in each block.